TrainingManagementSystem/src/main/java/trainingManagementSystem/dao/UserDao.java

```diff
@@ -38,6 +43,80 @@ public void updateUser(User user) {
 		hibernateTemplate.update(user);
 	}
 
+	// get users in a division
+	public List<User> getUserByDivisionId(int divisionId) {
+		Session session = hibernateTemplate.getSessionFactory().openSession();
+
+		CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+		CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+		Root<User> root = criteriaQuery.from(User.class);
+		CriteriaQuery<User> selectQuery = criteriaQuery.select(root);
+		selectQuery = selectQuery.where(criteriaBuilder.equal(root.get("division"), divisionId));
+		TypedQuery<User> typedQuery = session.createQuery(selectQuery);
+		List<User> users = typedQuery.getResultList();
+
+		session.close();
+		return users;
+	}
+
+	// [manager] remove user from a division
+	public void removeUserFromDivision(int id) {
+		try {
+			Session session = hibernateTemplate.getSessionFactory().openSession();
+			Transaction tx = session.beginTransaction();
+
+			User user = (User) session.get(User.class, id);
+
+			user.setDivision(null);
```

Comment:
Nếu user đó là admin của division thì sao em?

Reply:
Dạ oke ạ. Nghĩ lại thì một số chỗ em cũng chưa xử lý phần role đấy ạ. Em sẽ check lại ạ

Reply:
À nhưng chị ơi, phần này nếu về giao diện thì manager khi xem ds nhân viên thì sẽ không bao gồm chính mình nên không thể tự xóa chính mình được. Còn về BE thì tạm thời chắc em vẫn để lại để đợi phần phân quyền của anh Bình.